packages/adapter-drizzle/src/mysql/index.ts

```diff
@@ -0,0 +1,254 @@
+  email: varchar("email", { length: 255 }).notNull(),
+  emailVerified: timestamp("emailVerified", { mode: "date" }),
+  image: varchar("image", { length: 255 }),
+})
+
+export const accounts = mysqlTable(
+  "accounts",
+  {
+    userId: varchar("userId", { length: 255 })
+      .notNull()
+      .references(() => users.id, { onDelete: "cascade" }),
+    type: varchar("type", { length: 255 }).$type<AdapterAccount["type"]>().notNull(),
+    provider: varchar("provider", { length: 255 }).notNull(),
+    providerAccountId: varchar("providerAccountId", { length: 255 }).notNull(),
+    refresh_token: varchar("refresh_token", { length: 255 }),
+    access_token: varchar("access_token", { length: 255 }),
+    expires_at: int("expires_at"),
+    token_type: varchar("token_type", { length: 255 }),
+    scope: varchar("scope", { length: 255 }),
+    id_token: varchar("id_token", { length: 255 }),
+    session_state: varchar("session_state", { length: 255 }),
+  },
+  (account) => ({
+    compoundKey: primaryKey(account.provider, account.providerAccountId),
+  })
+)
+
+export const sessions = mysqlTable("sessions", {
+  sessionToken: varchar("sessionToken", { length: 255 }).notNull().primaryKey(),
+  userId: varchar("userId", { length: 255 })
+    .notNull()
+    .references(() => users.id, { onDelete: "cascade" }),
+  expires: timestamp("expires", { mode: "date" }).notNull(),
+})
+
+export const verificationTokens = mysqlTable(
+  "verificationToken",
+  {
+    identifier: varchar("identifier", { length: 255 }).notNull(),
+    token: varchar("token", { length: 255 }).notNull(),
+    expires: timestamp("expires", { mode: "date" }).notNull(),
+  },
+  (vt) => ({
+    compoundKey: primaryKey(vt.identifier, vt.token),
+  })
+)
+
+const poolConnection = process.env.AUTH_DRIZZLE_MYSQL_URL ? createPool(
+  process.env.AUTH_DRIZZLE_MYSQL_URL as string,
+) : createPool({
+  host: "localhost",
+  user: "root",
+  password: "password",
+  database: "next-auth",
+})
+
+export const db = drizzle(poolConnection)
+export type DbClient = typeof db
+
+export const defaultSchema = { users, accounts, sessions, verificationTokens }
+export type DefaultSchema = typeof defaultSchema
+interface CustomSchema extends DefaultSchema { }
+
+export function mySqlDrizzleAdapter(
+  client: DbClient,
+  schema?: Partial<CustomSchema>
+): Adapter {
+  // const { users, accounts, sessions, verificationTokens } = {
+  //   users: schema?.users ?? defaultSchema.users,
+  //   accounts: schema?.accounts ?? defaultSchema.accounts,
+  //   sessions: schema?.sessions ?? defaultSchema.sessions,
+  //   verificationTokens:
+  //     schema?.verificationTokens ?? defaultSchema.verificationTokens,
+  // }
+
+  return {
+    createUser: async (data) => {
+      const id = crypto.randomUUID()
+
+      await client.insert(users).values({ ...data, id })
+
+      return client
+        .select()
+        .from(users)
+        .where(eq(users.id, id))
+        .then((res) => res[0])
```

Comment:
This is giving me a typescript error when trying to pass the adapter to `nextauth` as Adapter

```
Type 'Adapter' is not assignable to type 'Adapter | undefined'.
  Type 'Adapter' is not assignable to type 'DefaultAdapter & { createVerificationToken: (verificationToken: VerificationToken) => Awaitable<VerificationToken | null | undefined>; useVerificationToken: (params: { ...; }) => Awaitable<...>; }'.
    Type 'Adapter' is not assignable to type 'DefaultAdapter'.
      Types of property 'createUser' are incompatible.
        Type '((user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>) | undefined' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.
          Type 'undefined' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.

```

Reply:
I fixed the error by lying Typescript, not even tried yet. But I would like to understand why `.where` is giving `undefined` in Drizzel. Please comment if you know why : )

<img width="540" alt="image" src="https://github.com/nextauthjs/next-auth/assets/49499/1b9e6d79-2cea-43ad-a2b8-2452cef9baa8">